internal/cmd/controller/reconciler/cronduration_job_test.go

```diff
@@ -0,0 +1,546 @@
+
+			timeBeforeSchedule := time.Now()
+			err = job.scheduleJob(ctx)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Check if job is scheduled
+			scheduledJob, err := scheduler.GetScheduledJob(job.key)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(scheduledJob).NotTo(BeNil())
+
+			// Check schedule status
+			updatedSchedule := &fleet.Schedule{}
+			err = k8sclient.Get(ctx, types.NamespacedName{Name: "test-schedule", Namespace: "default"}, updatedSchedule)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(updatedSchedule.Status.Active).To(BeFalse())
+			// time should be the next minute and 0 seconds
+			Expect(updatedSchedule.Status.NextStartTime.Time.Second()).To(BeZero())
+			Expect(updatedSchedule.Status.NextStartTime.Time.Minute()).To(Equal(timeBeforeSchedule.Add(time.Minute).Minute()))
+			Expect(updatedSchedule.Status.NextStartTime.Time.Hour()).To(Equal(timeBeforeSchedule.Hour()))
+			// Expect(updatedSchedule.Status.NextStartTime.Time).To(BeTemporally("~", timeBeforeSchedule.Add(time.Minute), 2*time.Second))
+			Expect(updatedSchedule.Status.MatchingClusters).To(Equal(job.MatchingClusters))
+		})
+	})
+
+	Context("updateJob", func() {
+		It("should update an existing job", func() {
+			job, err := NewCronDurationJob(ctx, schedule, scheduler, k8sclient)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = job.scheduleJob(ctx)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Modify schedule to force an update
+			schedule.Spec.Schedule = "0 */2 * * * *"
+			updatedJob, err := NewCronDurationJob(ctx, schedule, scheduler, k8sclient)
+			Expect(err).NotTo(HaveOccurred())
+
+			timeBeforeUpdate := time.Now()
+			err = updatedJob.updateJob(ctx)
+			Expect(err).NotTo(HaveOccurred())
+
+			scheduledJob, err := scheduler.GetScheduledJob(job.key)
+			Expect(err).NotTo(HaveOccurred())
+			cronDurationJob, ok := scheduledJob.JobDetail().Job().(*CronDurationJob)
+			Expect(ok).To(BeTrue())
+			Expect(cronDurationJob.Schedule).To(Equal(schedule))
+
+			// Check schedule status
+			updatedSchedule := &fleet.Schedule{}
+			err = k8sclient.Get(ctx, types.NamespacedName{Name: "test-schedule", Namespace: "default"}, updatedSchedule)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(updatedSchedule.Status.Active).To(BeFalse())
+
+			// next fire time should be the next even minute and 0 seconds
+			Expect(updatedSchedule.Status.NextStartTime.Time.Second()).To(BeZero())
+			if timeBeforeUpdate.Minute()%2 == 0 {
+				Expect(updatedSchedule.Status.NextStartTime.Time.Minute()).To(Equal(timeBeforeUpdate.Add(2 * time.Minute).Minute()))
+			} else {
+				Expect(updatedSchedule.Status.NextStartTime.Time.Minute()).To(Equal(timeBeforeUpdate.Add(1 * time.Minute).Minute()))
+			}
+			if timeBeforeUpdate.Minute() < 59 {
+				Expect(updatedSchedule.Status.NextStartTime.Time.Hour()).To(Equal(timeBeforeUpdate.Hour()))
+			} else {
+				Expect(updatedSchedule.Status.NextStartTime.Time.Hour()).To(Equal(timeBeforeUpdate.Add(1 * time.Hour).Hour()))
```

Comment:
Again, at the risk of being picky, what if this test case starts running at 23:59? :upside_down_face: